### ISSUE TITLE
[Improvement] - nomad run job.nomad -var foo=bar

### ISSUE THREAD

**Issue by samber** [hashicorp/nomad#2422]
Hi guys,

In terraform I use pretty often the ```-var``` and ```-var-file``` parameters, to send some variable into the .tf.

What about providing the same kind of feature for nomad ?

my-awesome-api.nomad:
```
...
    task "my-awesome-api" {
      driver = "docker"

      config {
        image = "acme/my-awesome-api:${var.API_VERSION}"
      }
...
```

then
```
$ nomad run my-awesome-api.nomad -var API_VERSION=4.2
```

**Comment by dradtke** [comment]
For anyone who stumbles across this issue, here's some more context on this feature from a Nomad dev: https://news.ycombinator.com/item?id=13843071

**Comment by Miserlou** [comment]
This is pretty critical for anybody using Nomad in a team. We'd also like to be able to pass in this variables from a file, or as environment variables.

**Comment by rkettelerij** [comment]
For anyone else who has a need for this: We're currently using Terraform as a frontend for Nomad in order to parameterise/template Nomad jobs. Basically the approach outlined in https://www.hashicorp.com/blog/continuous-deployment-with-nomad-and-terraform. 

This works fine, but nevertheless native support in Nomad for job parameterisation and templating is much appreciated.

**Comment by stevenscg** [comment]
I can't recommend [Levant](https://github.com/jrasell/levant) highly enough for this kind of task.

I am actually moving all of my Nomad jobs and deployment processes to it right now.

**Comment by rkettelerij** [comment]
> We're currently using Terraform as a frontend for Nomad in order to parameterise/template Nomad jobs. Basically the approach outlined in https://www.hashicorp.com/blog/continuous-deployment-with-nomad-and-terraform.

To get back on my comment. It's a few months later now and altough Terraform as a template tool for Nomad jobs works. I found [Levant](https://github.com/jrasell/levant) to be a superior alternative. Above all because Levant is able to monitor your Nomad deployment which in turn allows you to pass/fail your CI/CD pipeline. So for future readers in need of a templating solution: try Levant.

**Comment by tgross** [comment]
Shipped in Nomad 1.0 with our HCL2 support.
